### ISSUE TITLE
Expected length "-Infinity"

### ISSUE THREAD

**Issue by Eugenitals** [jsplumb/jsPlumb#607]
Hello!

I looked into some of your source code, and realized that in case of using 'Straight' connector it is possible to get unhandled errors. 
Here (https://github.com/jsplumb/jsPlumb/blob/master/src/connectors-straight.js#L28) in case when connector length is zero (for example while dragging new connection) due to this (https://github.com/jsplumb/jsPlumb/blob/master/src/defaults.js#L574) check, and because couple of lines above we have resetted (https://github.com/jsplumb/jsPlumb/blob/master/src/connection.js#L440) bounds into infinitives and then called `calculate` (https://github.com/jsplumb/jsPlumb/blob/master/src/defaults.js#L688) function, connector bounds will not be re-calculated during `compute` function call at all.

This cause to errors like this:
```
jsplumb.js:13036 Error: <svg> attribute width: Expected length, "-Infinity".
jsplumb.js:13036 Error: <svg> attribute height: Expected length, "-Infinity".
```

This issue occurs when you starting to drag a new connection and move cursor very close to source endpoint i.e. when connection length is zero.

**Comment by BMaxV** [comment]
I think I have a related warning, when connecting two nodes programmatically I get

"Unexpected value -Infinity parsing width attribute. "

Although only in some cases, I haven't figured out how to reproduce that.

**Comment by Lucas-C** [comment]
I have the same issue,
and also did Edward Wong a year ago: https://groups.google.com/forum/#!topic/jsplumb/PuDfP5QbQfc

**Comment by andrewfan** [comment]
@Eugenitals @Lucas-C did you find any workarounds for this issue?

**Comment by andrewfan** [comment]
Speaking about reproduction 
you can use this demo https://jsplumbtoolkit.com/demos/wrapped/draggableConnectors/index.html
for example, click on the green dot and try to slightly drag it to the top.

**Comment by Eugenitals** [comment]
@andrewfan No i didn't. But actually it didn't break my code, just spam in console.

**Comment by nxhoang** [comment]
this issue hasn't been resolved yet, has it?

**Comment by wuhduhren** [comment]
1. The reason of the problem is because you addNode and addEdge before rendering, and those endpoint are overlapping.
2. So you need to put those addNode, addEdge in toolkit.batch(...).
3. batch() temporary stops rendering, do things in batch(),  and refresh afterwards. Just search for batch() in jsplumb you will understand.

**Comment by sporritt** [comment]
hmm but those method calls are in the Toolkit edition, @wuhduhren .

**Comment by wuhduhren** [comment]
Yeah, I am using the Toolkit. But I saw batch in community edition's document too.
https://jsplumbtoolkit.com/community/doc/home.html
(search: batch

**Comment by thenewbeat** [comment]
I'm using Community, and the `batch()` option doesn't not stop this error from happening in the console.

**Comment by baibhabmondal** [comment]
Still facing this error.
